src/Catalyst.Modules.Server.Blazor/AutofacServiceProvider.cs

```diff
@@ -0,0 +1,137 @@
+        /// </summary>
+        /// <param name="containerBuilder">The container builder.</param>
+        /// <returns>An <see cref="IServiceProvider" />.</returns>
+        public IServiceProvider CreateServiceProvider(ContainerBuilder containerBuilder)
+        {
+            if (containerBuilder == null) throw new ArgumentNullException(nameof(containerBuilder));
+
+            return _autofacServiceProvider = new AutofacServiceProvider();
+        }
+    }
+
+    /// <summary>
+    /// Autofac implementation of the ASP.NET Core <see cref="IServiceProvider"/>.
+    /// </summary>
+    /// <seealso cref="System.IServiceProvider" />
+    /// <seealso cref="Microsoft.Extensions.DependencyInjection.ISupportRequiredService" />
+    internal class AutofacServiceProvider : IServiceProvider, ISupportRequiredService, IDisposable
+    {
+        public ILifetimeScope LifetimeScope { get; set; }
+
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Gets service of type <paramref name="serviceType" /> from the
+        /// <see cref="AutofacServiceProvider" /> and requires it be present.
+        /// </summary>
+        /// <param name="serviceType">
+        /// An object that specifies the type of service object to get.
+        /// </param>
+        /// <returns>
+        /// A service object of type <paramref name="serviceType" />.
+        /// </returns>
+        /// <exception cref="Autofac.Core.Registration.ComponentNotRegisteredException">
+        /// Thrown if the <paramref name="serviceType" /> isn't registered with the container.
+        /// </exception>
+        /// <exception cref="Autofac.Core.DependencyResolutionException">
+        /// Thrown if the object can't be resolved from the container.
+        /// </exception>
+        public object GetRequiredService(Type serviceType) { return LifetimeScope.Resolve(serviceType); }
+
+        /// <summary>
+        /// Gets the service object of the specified type.
+        /// </summary>
+        /// <param name="serviceType">
+        /// An object that specifies the type of service object to get.
+        /// </param>
+        /// <returns>
+        /// A service object of type <paramref name="serviceType" />; or <see langword="null" />
+        /// if there is no service object of type <paramref name="serviceType" />.
+        /// </returns>
+        public object GetService(Type serviceType) { return LifetimeScope.ResolveOptional(serviceType); }
+
+        /// <summary>
+        /// Releases unmanaged and - optionally - managed resources.
+        /// </summary>
+        /// <param name="disposing">
+        /// <see langword="true" /> to release both managed and unmanaged resources;
+        /// <see langword="false" /> to release only unmanaged resources.
+        /// </param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing) LifetimeScope.Dispose();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)